lib/commons/text/unicode.js

```diff
@@ -130,11 +130,14 @@ function getPunctuationRegExp() {
  * @returns {RegExp}
  */
 function getSupplementaryPrivateUseRegExp() {
+	// Supplementary private use area A (https://www.unicode.org/charts/PDF/UF0000.pdf) contains
+	// characters between F0000 and FFFFF. Because ES5 doesn't have a syntax for regular expressions
+	// of such characters, search instead for the corresponding surrogate pairs.
+	//
 	// 1. High surrogate area (https://www.unicode.org/charts/PDF/UD800.pdf)
 	// 2. Low surrogate area (https://www.unicode.org/charts/PDF/UDC00.pdf)
-	// 3. Supplementary private use area A (https://www.unicode.org/charts/PDF/UF0000.pdf)
 	//
-	//             1              2                                  3
-	//      ┏━━━━━━┻━━━━━━┓┏━━━━━━┻━━━━━━┓ ┏━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
-	return /[\uDB80-\uDBBF][\uDC00-\uDFFD]|(?:[\uDB80-\uDBBE][\uDC00-\uDFFF]|\uDBBF[\uDC00-\uDFFD])/g;
+	//             1              2
+	//      ┏━━━━━━┻━━━━━━┓┏━━━━━━┻━━━━━━┓
+	return /[\uDB80-\uDBBF][\uDC00-\uDFFF]/g;
```

Comment:
Note that the low surrogate has been expanded to include DFFF. Without that, I think we exclude characters such as [F03FF (\uDB80\uDFFF)](https://www.fileformat.info/info/unicode/char/f03ff/index.htm), which seems to be a valid character in the private use area. I've updated the tests to catch this.

Technically, FFFFE and FFFFF are in the private use area, but are "non characters".

I think it's okay to still match those, but if we think that will be an issue, we can replace this with the part I originally added in #2102:

```js
return /[\uDB80-\uDBBE][\uDC00-\uDFFF]|\uDBBF[\uDC00-\uDFFD]/g;
```

which would exclude FFFFE/FFFFF.

Reply:
I think it should be ok if we include FFFFE and FFFFF. The regex is used to strip out characters that we shouldn't check for color contrast so non characters fit that bill.

Reply:
Could you add this as a comment to the code? That way when we look at this in the future we'll remember why the low surrogate doesn't end at uDFFD (which the doc points to)?

Reply:
Will do

Reply:
And done